eclair-node/src/main/scala/fr/acinq/eclair/api/handlers/Channel.scala

```diff
@@ -0,0 +1,98 @@
+import fr.acinq.eclair.api.Service
+import fr.acinq.eclair.api.directives.EclairDirectives
+import fr.acinq.eclair.blockchain.fee.FeeratePerByte
+import scodec.bits.ByteVector
+
+trait Channel {
+  this: Service with EclairDirectives =>
+
+  import fr.acinq.eclair.api.serde.JsonSupport.{formats, marshaller, serialization}
+
+  val open: Route = postRequest("open") { implicit t =>
+    formFields(nodeIdFormParam, "fundingSatoshis".as[Satoshi], "pushMsat".as[MilliSatoshi].?, "fundingFeerateSatByte".as[FeeratePerByte].?, "feeBaseMsat".as[MilliSatoshi].?,
+      "feeProportionalMillionths".as[Int].?, "channelFlags".as[Int].?, "openTimeoutSeconds".as[Timeout].?) {
+      (nodeId, fundingSatoshis, pushMsat, fundingFeerateSatByte, feeBase, feeProportional, channelFlags, openTimeout_opt) =>
+        if (feeBase.nonEmpty && feeProportional.isEmpty || feeBase.isEmpty && feeProportional.nonEmpty) {
+          reject(MalformedFormFieldRejection("feeBaseMsat/feeProportionalMillionths",
+            "All relay fees parameters (feeBaseMsat/feeProportionalMillionths) must be specified to override node defaults"
+          ))
+        } else {
+          val initialRelayFees = (feeBase, feeProportional) match {
+            case (Some(feeBase), Some(feeProportional)) => Some(feeBase, feeProportional)
+            case _ => None
+          }
+          complete {
+            eclairApi.open(nodeId, fundingSatoshis, pushMsat, fundingFeerateSatByte, initialRelayFees, channelFlags, openTimeout_opt)
+          }
+        }
+    }
+  }
+
+  val close: Route = postRequest("close") { implicit t =>
+    withChannelsIdentifier { channels =>
+      formFields("scriptPubKey".as[ByteVector](binaryDataUnmarshaller).?) { scriptPubKey_opt =>
+        complete(eclairApi.close(channels, scriptPubKey_opt))
+      }
+    }
+  }
+
+  val forceClose: Route = postRequest("forceclose") { implicit t =>
+    withChannelsIdentifier { channels =>
+      complete(eclairApi.forceClose(channels))
+    }
+  }
+
+  val channel: Route = postRequest("channel") { implicit t =>
+    withChannelIdentifier { channel =>
+      complete(eclairApi.channelInfo(channel))
+    }
+  }
+
+  val channels: Route = postRequest("channels") { implicit t =>
+    formFields(nodeIdFormParam.?) { toRemoteNodeId_opt =>
+      complete(eclairApi.channelsInfo(toRemoteNodeId_opt))
+    }
+  }
+
+  val allChannels: Route = postRequest("allchannels") { implicit t =>
+    complete(eclairApi.allChannels())
+  }
+
+  val allUpdates: Route = postRequest("allupdates") { implicit t =>
+    formFields(nodeIdFormParam.?) { nodeId_opt =>
+      complete(eclairApi.allUpdates(nodeId_opt))
+    }
+  }
+
+  val channelStats: Route = postRequest("channelstats") { implicit t =>
+    formFields(fromFormParam.?, toFormParam.?) { (from_opt, to_opt) =>
+      complete(eclairApi.channelStats(from_opt, to_opt))
+    }
+  }
+
+  val channelRoutes: Route = open ~ close ~ forceClose ~ channel ~ channels ~ allChannels ~ allUpdates ~ channelStats
```

Comment:
Defining a variable for each single path seems a bit overkill, and now we must not forget to add each one of them to the `channelRoutes`.

Reply:
It would be very easy to collect them in a block (you still need to chain them), you loose the ability to use/test them independently though and they will be intended again (but just 1 level).

Reply:
Fair enough.